moveos/moveos-stdlib/src/natives/moveos_stdlib/move_module.rs

```diff
@@ -270,22 +260,80 @@ fn check_compatibililty_inner(
+ * ): vector<vector<u8>>;
  * Native function to remap addresses in module binary where the length of
  * `old_addresses` must equal to that of `new_addresses`.
  **************************************************************************************************/
 #[derive(Debug, Clone)]
-pub struct RemapAddressesGasParameters {
+pub struct ReplaceAddressIdentifierGasParameters {
+    pub base: InternalGas,
+    pub per_byte: InternalGasPerByte,
+}
+
+fn replace_address_identifiers(
+    gas_params: &ReplaceAddressIdentifierGasParameters,
+    _context: &mut NativeContext,
+    _ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(args.len() == 3, "Wrong number of arguments");
+    let mut cost = gas_params.base;
+    let new_address_vec = pop_arg!(args, Vector);
+    let old_address_vec = pop_arg!(args, Vector);
+    let num_addresses = new_address_vec.elem_views().len();
+    if num_addresses != old_address_vec.elem_views().len() {
+        return Ok(NativeResult::err(
+            cost,
+            moveos_types::move_std::error::invalid_argument(E_LENTH_NOT_MATCH),
+        ));
+    };
+    let num_addresses = num_addresses as u64;
+    let new_addresses = new_address_vec.unpack(&Type::Address, num_addresses)?;
+    let old_addresses = old_address_vec.unpack(&Type::Address, num_addresses)?;
+
+    let address_mapping: HashMap<AccountAddress, AccountAddress> =
+        zip_eq(old_addresses, new_addresses)
+            .map(|(a, b)| {
+                Ok((
+                    a.value_as::<AccountAddress>()?,
+                    b.value_as::<AccountAddress>()?,
+                ))
+            })
+            .collect::<PartialVMResult<_>>()?;
+
+    let mut bundle = vec![];
+    for module in pop_arg!(args, Vec<Value>) {
+        let byte_codes = module.value_as::<Vec<u8>>()?;
+        cost += gas_params.per_byte * NumBytes::new(byte_codes.len() as u64);
+        bundle.push(byte_codes);
+    }
+    let output_modules = modify_modules(bundle, |module| {
+        module_replace_address_identifiers(module, &address_mapping)
+    })?;
+    Ok(NativeResult::ok(cost, smallvec![output_modules]))
+}
+
+/***************************************************************************************************
+ * native fun replace_addresses_constant(
+ *     bytes: vector<vector<u8>>,
+ *     old_addresses: vector<address>,
+ *     new_addresses: vector<address>,
+ * ): (vector<u8>, vector<vector<u8>>);
+ * Native function to replace constant addresses in module binary where the length of
+ * `old_addresses` must equal to that of `new_addresses`.
+ **************************************************************************************************/
+#[derive(Debug, Clone)]
+pub struct ReplaceAddressConstantGasParameters {
     pub base: InternalGas,
     pub per_byte: InternalGasPerByte,
 }
 
-fn remap_module_addresses_inner(
-    gas_params: &RemapAddressesGasParameters,
+fn replace_addresses_constant(
```

Comment:
What is the different between `replace_addresses_constant ` and `replace_address_identifiers `?

Reply:
The native implementations of `replace_addresses_constant` and `replace_address_identifiers` are almost the same. Perhaps abstraction and code reuse can be considered in the next PR.